src/js/components/productSupplier/PreferenceTypeColumn.jsx

```diff
@@ -16,20 +18,49 @@ const getLabel = (productSupplierPreferences) => {
     return {
       id: 'react.productSupplier.preferenceType.multiple.label',
       defaultMessage: 'Multiple',
+      icon: <RiInformationLine />,
+      className: 'cell-content',
     };
   }
   return productSupplierPreferences[0].preferenceType?.id;
 };
 
-const PreferenceTypeColumn = ({ productSupplierPreferences }) => {
+const PreferenceTypeColumn = ({ productSupplierPreferences, productSupplierId }) => {
+  const [preferenceTypeModalData, setPreferenceTypeModalData] = useState([]);
   const label = getLabel(productSupplierPreferences);
 
+  const onCellClick = () => {
+    if (productSupplierPreferences.length > 1) {
+      setPreferenceTypeModalData(productSupplierPreferences);
+    }
+  };
+
+  const closeModal = () => setPreferenceTypeModalData([]);
+
   return (
-    <span>
-      {_.isObject(label)
-        ? <Translate id={label.id} defaultMessage={label.defaultMessage} />
-        : label}
-    </span>
+    <>
+      <span
+        className={label?.className}
+        onClick={onCellClick}
```

Comment:
here it doesn't matter, because the definition of your method is `() => ...`, but be aware that if you had an argument, it would be called with the `e` from `span`.
I mean that your current code evaluets to:
```
onClick={e => onCellClick(e)}
```